classes/submission/form/PKPSubmissionSubmitStep1Form.inc.php

```diff
@@ -161,13 +161,60 @@ function setSubmissionData($submission) {
 	}
 
 	/**
+	 * Add or update comments to editor
+	 */
+	function setCommentsToEditor($submissionId, $commentsToEditor, $userId, $update = null) {
+		$queryDao = DAORegistry::getDAO('QueryDAO');
+		$noteDao = DAORegistry::getDAO('NoteDAO');
+
+		if (!isset($update)){
+			$query = $queryDao->newDataObject();
+			$query->setAssocType(ASSOC_TYPE_SUBMISSION);
+			$query->setAssocId($submissionId);
+			$query->setStageId(WORKFLOW_STAGE_ID_SUBMISSION);
+			$query->setSequence(REALLY_BIG_NUMBER);
+			$queryDao->insertObject($query);
+			$queryDao->resequence(ASSOC_TYPE_SUBMISSION, $submissionId);
+			$queryDao->insertParticipant($query->getId(), $userId);
+			$queryId = $query->getId();
+
+			$note = $noteDao->newDataObject();
+			$note->setUserId($userId);
+			$note->setAssocType(ASSOC_TYPE_QUERY);
+			$note->setTitle('Comments to editor'); // Localize?
+			$note->setContents($commentsToEditor);
+			$note->setDateCreated(Core::getCurrentDate());
+			$note->setDateModified(Core::getCurrentDate());			
+			$note->setAssocId($queryId);
+			$noteDao->insertObject($note);
+
+		} else{
+			$queries = $queryDao->getByAssoc(ASSOC_TYPE_SUBMISSION, $submissionId);
+			$query = $queries->next();
+			$queryId = $query->getData('id');
```

Comment:
I think I had difficulties in getting getId() to work there, I am not sure why. I will try again.